src/MultiFormats/VarInt.cs

```diff
@@ -0,0 +1,264 @@
+        /// <exception cref="InvalidDataException">
+        ///   When the varint value is greater than <see cref="Int64.MaxValue"/>.
+        /// </exception>
+        /// <returns>The integer value.</returns>
+        public static long ReadVarint64(this Stream stream)
+        {
+            return stream.ReadVarint64Async().ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        ///   Writes the variable integer encoding of the value `to
+        ///   a stream.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to write to.
+        /// </param>
+        /// <param name="value">
+        ///   A non-negative value to write.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation.
+        /// </returns>
+        /// <exception cref="NotSupportedException">
+        ///   When <paramref name="value"/> is negative.
+        /// </exception>
+        public static async Task WriteVarintAsync(this Stream stream,
+            long value,
+            CancellationToken cancel = default)
+        {
+            if (value < 0) throw new NotSupportedException("Negative values are not allowed for a Varint");
+
+            var bytes = new byte[10];
+            var i = 0;
+            do
+            {
+                var v = (byte) (value & 0x7F);
+                if (value > 0x7F) v |= 0x80;
+
+                bytes[i++] = v;
+                value >>= 7;
+            } while (value != 0);
+
+            await stream.WriteAsync(bytes, 0, i, cancel).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        ///   Reads a variable integer from the stream. 
+        /// </summary>
+        /// <param name="stream">
+        ///   A varint encoded <see cref="Stream"/>.
+        /// </param>
+        /// <param name="cancel">
+        ///   Is used to stop the task.  When cancelled, the <see cref="TaskCanceledException"/> is raised.
+        /// </param>
+        /// <returns>
+        ///   A task that represents the asynchronous operation. The task's result
+        ///   is the integer value in the <paramref name="stream"/>.
+        /// </returns>
+        /// <exception cref="EndOfStreamException">
+        ///   When the no bytes exist in the <paramref name="stream"/>.
+        /// </exception>
+        /// <exception cref="InvalidDataException">
+        ///   When the varint value is greater than <see cref="Int32.MaxValue"/>.
+        /// </exception>
+        public static async Task<int> ReadVarint32Async(this Stream stream,
+            CancellationToken cancel = default(CancellationToken))
+        {
+            var value = await stream.ReadVarint64Async(cancel).ConfigureAwait(false);
+            if (value > int.MaxValue)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)